schema.graphql

```diff
@@ -929,3 +932,146 @@ type ProjectExternalAssetDependency @entity {
   "The asset data if it is onchain"
   data: String
 }
+
+enum PMPAuthOptions {
+  "Artist has authority to configure PMP"
+  Artist,
+
+  "Token owner has authority to configure PMP"
+  TokenOwner,
+
+  "Contract address has authority to configure PMP"
+  Address,
+
+  "Artist and token owner have authority to configure PMP"
+  ArtistAndTokenOwner,
+
+  "Artist and contract address have authority to configure PMP"
+  ArtistAndAddress,
+
+  "Token owner and contract address have authority to configure PMP"
+  TokenOwnerAndAddress,
+
+  "Artist, token owner, and contract address have authority to configure PMP"
+  ArtistAndTokenOwnerAndAddress
+}
+
+enum PMPParamTypes {
+  "Unconfigured by default, used to check if PMP is configured"
+  Unconfigured,
+
+  "List of options"
+  Select,
+
+  "Boolean value"
+  Bool,
+
+  "Uint256 range"
+  Uint256Range,
+
+  "Int256 range"
+  Int256Range,
+
+  "Decimal range"
+  DecimalRange,
+
+  "Hex color"
+  HexColor,
+
+  "Timestamp"
+  Timestamp,
+
+  "String"
+  String
+}
+
+type PMPProjectConfig @entity {
+  "Unique identifier made up of {PMP_Plugin_Address}-{ProjectId}"
+  id: ID!
+
+  "Project associated with this PMP project config"
+  project: Project!
+
+  "Count of PMP configs for this project"
+  pmpConfigCount: BigInt!
+
+  "Array of keys for the current PMP configs"
+  pmpConfigKeys: [String!]!
+
+  "Timestamp of last update"
+  updatedAt: BigInt!
+
+  "List of PMP configs for this project"
+  PMPConfigs: [PMPConfig!]! @derivedFrom(field: "PMPProjectConfig")
+}
+
+type PMPConfig @entity {
+  "Unique identifier made up of {PMP_Plugin_Address}-{ProjectId}-{Key}"
+  id: ID!
+
+  "PMP Project config associated with this PMP config"
+  pmpProjectConfig: PMPProjectConfig!
+
+  "Auth option for this PMP config"
+  authOption: PMPAuthOptions!
+
+  "Param type for this PMP config"
+  paramType: PMPParamTypes!
+
+  "Parameter key for this PMP config"
+  key: String!
+
+  "Timestamp after which updates are no longer possible to the PMP"
+  pmpLockedAfterTimestamp: BigInt!
+
+  "only populated if auth is type Address, null otherwise"
+  authAddress: Bytes
+
+  "options to choose from, if Select type"
+  selectOptions: [String!]!
+
+  "bytes32 representation of type's minimum allowed value, if applicible"
+  minRange: String
+
+  "bytes32 representation of type's maximum allowed value, if applicible"
+  maxRange: String
+
+  "Timestamp of creation"
+  createdAt: BigInt!
+}
+
+type PMP @entity(immutable: true) {
+  "Unique identifier made up of {PMP_Plugin_Address}-{tokenId}-{Key}-{tokenPMPNonce}"
+  id: ID!
+
+  "The key for this PMP"
+  key: String!
+
+  "The token associated with this PMP"
+  token: Token!
```

Comment:
@lindsgil should we at an @derivedFrom on the token entity to establish the array relationship?

Reply:
🙏 🙏  added!